pkg/helmdeployer/deployer.go

```diff
@@ -372,10 +365,6 @@ func (h *helm) getValues(options fleet.BundleDeploymentOptions, defaultNamespace
 		values = options.Helm.Values.Data
 	}
 
-	if h.secretCache == nil || h.configmapCache == nil {
-		return values, nil
-	}
-
```

Comment:
Changed since last review: this is likely the culprit for #384. This check is not necessary since we should check `valuesFrom` as well.

We also never have to initialize this cache nor have to check whether or not it's `nil`, similar to `serviceAccountCache` in this same file. These caches now reflect `serviceAccountCache`'s behavior.